tests/end_to_end/end_to_end.cpp

```diff
@@ -537,61 +562,72 @@ TEST_CASE("bindmonitor-interpret", "[bindmonitor_interpret]")
+    program_information_provider_t bind_program_information(EBPF_PROGRAM_TYPE_BIND);
 
-    // Unbind first port
-    emulate_unbind(hook, fake_pid, "fake_app_1");
-    REQUIRE(get_bind_count_for_pid(map_handles[0], fake_pid) == 0);
+    REQUIRE(
+        (result = ebpf_api_load_program(
+             SAMPLE_PATH "bindmonitor.o",
+             "bind",
+             EBPF_EXECUTION_INTERPRET,
+             &program_handle,
+             &count_of_map_handles,
+             map_handles,
+             &error_message),
+         error_message ? printf("ebpf_api_load_program failed with %s\n", error_message) : 0,
+         ebpf_api_free_string(error_message),
+         error_message = nullptr,
+         result == EBPF_SUCCESS));
 
-    // Bind from two apps to test enumeration
-    REQUIRE(emulate_bind(hook, fake_pid, "fake_app_1") == BIND_PERMIT);
-    REQUIRE(get_bind_count_for_pid(map_handles[0], fake_pid) == 1);
+    single_instance_hook_t hook;
 
-    fake_pid = 54321;
-    REQUIRE(emulate_bind(hook, fake_pid, "fake_app_2") == BIND_PERMIT);
-    REQUIRE(get_bind_count_for_pid(map_handles[0], fake_pid) == 1);
+    std::string process_maps_name = "bindmonitor::process_maps";
+    std::string limit_maps_name = "bindmonitor::limits_map";
 
-    uint64_t pid;
-    REQUIRE(
-        ebpf_api_get_next_map_key(map_handles[0], sizeof(uint64_t), NULL, reinterpret_cast<uint8_t*>(&pid)) ==
-        EBPF_SUCCESS);
-    REQUIRE(pid != 0);
-    REQUIRE(
-        ebpf_api_get_next_map_key(
-            map_handles[0], sizeof(uint64_t), reinterpret_cast<uint8_t*>(&pid), reinterpret_cast<uint8_t*>(&pid)) ==
-        EBPF_SUCCESS);
-    REQUIRE(pid != 0);
-    REQUIRE(
-        ebpf_api_get_next_map_key(
-            map_handles[0], sizeof(uint64_t), reinterpret_cast<uint8_t*>(&pid), reinterpret_cast<uint8_t*>(&pid)) ==
-        ERROR_NO_MORE_ITEMS);
+    ebpf_handle_t handle_iterator = INVALID_HANDLE_VALUE;
+    REQUIRE(ebpf_api_get_next_map(handle_iterator, &handle_iterator) == EBPF_SUCCESS);
+    map_handles[2] = handle_iterator;
+    REQUIRE(ebpf_api_get_next_map(handle_iterator, &handle_iterator) == EBPF_SUCCESS);
+    map_handles[3] = handle_iterator;
+    REQUIRE(ebpf_api_get_next_map(handle_iterator, &handle_iterator) == EBPF_SUCCESS);
+    REQUIRE(handle_iterator == INVALID_HANDLE_VALUE);
 
-    hook.detach();
+    ebpf_map_definition_t map_definitions[_countof(map_handles)];
+    ebpf_map_definition_t process_map = {
+        sizeof(ebpf_map_definition_t), EBPF_MAP_TYPE_HASH, sizeof(uint64_t), sizeof(process_entry_t), 1024};
+
+    ebpf_map_definition_t limits_map = {
+        sizeof(ebpf_map_definition_t), EBPF_MAP_TYPE_ARRAY, sizeof(uint32_t), sizeof(uint32_t), 1};
+
+    for (size_t index = 0; index < _countof(map_handles); index++) {
+        REQUIRE(
+            ebpf_api_map_query_definition(
+                map_handles[index],
+                &map_definitions[index].size,
+                &map_definitions[index].type,
```

Comment:
error C2664: 'ebpf_result_t ebpf_api_map_query_definition(ebpf_handle_t,uint32_t *,uint32_t *,uint32_t *,uint32_t *,uint32_t *)': cannot convert argument 3 from 'ebpf_map_type_t *' to 'uint32_t *' [D:\a\ebpf-for-windows\ebpf-for-windows\tests\end_to_end\end_to_end.vcxproj]